courseware/api.py

```diff
@@ -182,21 +186,50 @@ def _create_tokens_and_update_auth(auth, params):
     return auth
 
 
-@transaction.atomic
-def refresh_edx_api_auth(user):
+def get_valid_edx_api_auth(user, ttl_in_seconds=OPENEDX_AUTH_DEFAULT_TTL_IN_SECONDS):
     """
-    Updates the api tokens for the given user
+    Returns a valid api auth, possibly refreshing the tokens
 
     Args:
-        user (users.models.User): the user to update auth for
+        user (users.models.User): the user to get an auth for
+        ttl_in_seconds (int): number of seconds the auth credentials should still be valid
```

Comment:
(Take or leave) Possibly clearer: `How long the auth credentials need to remain unexpired without needing a refresh (in seconds)`